packages/next/src/server/lib/find-vercel-config.ts

```diff
@@ -0,0 +1,7 @@
+import findUp from 'next/dist/compiled/find-up'
+
+export async function hasVercelConfigFiles(dir: string) {
+  const vercelJsonPath = await findUp('vercel.json', { cwd: dir })
+  const dotVercelPath = await findUp('.vercel', { cwd: dir })
```

Comment:
What is the purpose of looking for `.vercel`? 

That might not exist until after running `next build`.